nf_core/commands_test_datasets.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import os
+
+import questionary
+import rich
+
+from nf_core.test_datasets.test_datasets_utils import (
+    MODULES_BRANCH_NAME,
+    create_download_url,
+    create_pretty_nf_path,
+    get_remote_branch_names,
+    list_files_by_branch,
+)
+from nf_core.utils import nfcore_question_style, rich_force_colors
+
+log = logging.getLogger(__name__)
+stdout = rich.console.Console(force_terminal=rich_force_colors())
+
+# Files / directories starting with one of the following in a git tree are ignored:
+IGNORED_FILE_PREFIXES = [
+    ".",
+    "CITATION",
+    "LICENSE",
+    "README",
+    "docs",
+]
+
+
+def test_datasets_list_branches(ctx):
+    """
+    List all branches on the nf-core/test-datasets repository.
+    Only lists test data and module test data based on the curated list
+    of pipeline names [on the website](https://raw.githubusercontent.com/nf-core/website/refs/heads/main/public/pipeline_names.json)
+    """
+    remote_branches = get_remote_branch_names()
+    out = os.linesep.join(remote_branches)
+    stdout.print(out)
+
+
+def test_datasets_list_remote(ctx, branch, generate_nf_path, generate_dl_url):
+    """
+    List all files on a given branch in the remote nf-core/testdatasets repository on github.
+    Specifying a branch is required.
+    The resulting files can be parsed as a nextflow path or a url for downloading.
+    """
+    tree = list_files_by_branch(branch, IGNORED_FILE_PREFIXES)
+    num_branches = len(tree.keys())
+
+    out = ""
+    for b in tree.keys():
+        branch_info = f"(Branch: {b})" if num_branches > 1 else ""
+        files = sorted(tree[b])
+        for f in files:
+            if generate_nf_path:
+                out += branch_info + create_pretty_nf_path(f, branch == MODULES_BRANCH_NAME) + os.linesep
+            elif generate_dl_url:
+                out += branch_info + create_download_url(branch, f) + os.linesep
+            else:
+                out += branch_info + f + os.linesep
```

Comment:
This code should be moved to a file `nf_core/test_datasets/list.py`, similar to what we do with `pipeliens list` for example